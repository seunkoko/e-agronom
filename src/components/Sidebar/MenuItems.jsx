// react library
import React from 'react';
import { Link } from 'react-router-dom';

// styles
import './Sidebar.scss';

/**
 * Renders MenuItems component
 * @param {object} props - component props
 * @returns {JSX} JSX
 */
const MenuItems = ({
  id,
  name
}) => (
  <Link
    to={`/${id}`}
    className="links"
  >
    <div className="menu-items">
      <i className="fas fa-cat menu-cat-icon"></i>
      <p className="menu-item-name">{name}</p>
    </div>
  </Link>
);

export default MenuItems;
