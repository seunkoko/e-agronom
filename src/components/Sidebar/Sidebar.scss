@import "../../stylesheets/variables";

.sidebar {
  position: fixed;
  background-color: $primary-black;
  height: 100vh;
  color: $primary-white;
  font-weight: bold;
  width: 20%;

  @media (max-width: $laptop-width) {
    width: 25%;
  }

  @media (max-width: $mobile-width) {
    width: 32%;
  }

  @media (max-width: $phone-width) {
    width: 20%;
  }
}

.sidebar-title {
  margin: 30px 20px;

  &:hover {
    cursor: pointer;
    color: $primary-orange;
  }
}

.menu-items{
  cursor: pointer;
  width: 100%;

  &:hover {
    background-color: $hover-orange;
    color: $primary-black;
  }
}

.menu-cat-icon {
  margin-left: 20px;
}

.menu-item-name {
  display: inline-block;
  margin-left: 10px;
}

.open-sidebar {
  cursor: pointer;
  margin: 30px 20px;
  font-size: 30px;
}
