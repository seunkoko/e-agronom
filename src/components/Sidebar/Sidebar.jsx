// react library
import React from 'react';

// components
import MenuItems from './MenuItems';

// styles
import './Sidebar.scss';

/**
 * Renders Sidebar component
 * @param {object} props - component props
 * @returns {JSX} JSX
 */
const Sidebar = ({
  openMobileMenu,
  categories
}) => (
  <div id="sidebar-id" className="sidebar">
    <div className="show-on-phone">
      <i
        id="open-sidebar-icon"
        className="fas fa-arrow-right open-sidebar"
        onClick={openMobileMenu}
      >
      </i>
    </div>

    <div id="sidebar-content" className="hide-on-phone">
      <h2 className="sidebar-title">
        CATEGORIES
      </h2>

      {
        categories && categories.map(category => {
          const item = <MenuItems
            key={category.id}
            id={category.id}
            name={category.name}
          />
          return item;
        })
      }
    </div>
  </div>
);

export default Sidebar;
