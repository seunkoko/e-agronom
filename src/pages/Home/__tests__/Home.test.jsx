// react library
import React from 'react';

// third-party librariess
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import { spy } from 'sinon';
import configureMockStore from 'redux-mock-store';

// components
import ConnectedHome, { Home } from '../Home';
import Sidebar from '../../../components/Sidebar/Sidebar';

const resolvePromise = () => Promise.resolve();

const mockStore = configureMockStore([thunk]);

let store = mockStore({
  categories: {
    data: [],
    error: {},
  },
  categoryImages: {
    data: [],
    error: {},
  },
});

let props = {
  match: {
    params: 1,
  },
}

let mountedWrapper;
let shallowWrapper;

describe('Home Component', () => {

  beforeEach(() => {
    mountedWrapper = mount(
      <Provider store={store}>
        <ConnectedHome
          {...props}
          getCategories={resolvePromise}
          getCategoryImages={resolvePromise}
        />
      </Provider>
    );
  
    shallowWrapper = shallow(
      <Home
        {...props}
        getCategories={resolvePromise}
        getCategoryImages={resolvePromise}
      />
    );
  });

  it('renders correctly', () => {
    const rendering = toJson(mountedWrapper);
    expect(rendering).toMatchSnapshot();
  });

  it('renders correctly when connected to the store', () => {
    expect(mountedWrapper.exists()).toBe(true);
  });

  it('should call handleGetCategoryImages method', () => {
    const handleGetCategoryImagesSpy = spy(shallowWrapper.instance(), 'handleGetCategoryImages');
    shallowWrapper.setState({
      categories: [],
      categoryImages: [],
      currentCategoryId: 1,
      currentPage: 1,
      pending: false,
      loadMorePending: false
    });
    shallowWrapper.instance().handleGetCategoryImages({
      categoryId: 5, page: 1,
    });
    expect(handleGetCategoryImagesSpy.called).toEqual(true);
  });

  it('should call handleLoadMore method', () => {
    shallowWrapper.setProps({ categoryImagesData: {
      data: [{ id: 1 }, { id: 2 }],
      error: {},
    } });

    const handleLoadMoreSpy = spy(shallowWrapper.instance(), 'handleLoadMore');
    shallowWrapper.setState({
      categories: [],
      categoryImages: {
        data: [{ id: 3 }, { id: 4 }],
        error: {},
      },
      currentCategoryId: 1,
      currentPage: 1,
      pending: false,
      loadMorePending: false
    });

    shallowWrapper.instance().handleLoadMore();

    expect(handleLoadMoreSpy.called).toEqual(true);
  });

  it('should simulate openMobileMenu function click event', () => {
    const openMobileMenuSpy = spy(shallowWrapper.instance(), 'openMobileMenu');

    const wrapper =  mount(
        <Sidebar
          categories={[]}
          openMobileMenu={openMobileMenuSpy}
        />, { attachTo: document.body }
    );

    const openMobileMenu = wrapper.find('#open-sidebar-icon');
    openMobileMenu.simulate('click');
    expect(wrapper.find('#sidebar-id').hasClass('mobile-style')).toEqual(false)
    expect(openMobileMenuSpy.called).toEqual(true);
  });
});
