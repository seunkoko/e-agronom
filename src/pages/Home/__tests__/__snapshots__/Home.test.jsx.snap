// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Home)
    getCategories={[Function]}
    getCategoryImages={[Function]}
    match={
      Object {
        "params": 1,
      }
    }
  >
    <Home
      getCategories={[Function]}
      getCategoryImages={[Function]}
      match={
        Object {
          "params": 1,
        }
      }
    >
      <div
        id="main-body"
      >
        <Sidebar
          openMobileMenu={[Function]}
        >
          <div
            className="sidebar"
            id="sidebar-id"
          >
            <div
              className="show-on-phone"
            >
              <i
                className="fas fa-arrow-right open-sidebar"
                id="open-sidebar-icon"
                onClick={[Function]}
              />
            </div>
            <div
              className="hide-on-phone"
              id="sidebar-content"
            >
              <h2
                className="sidebar-title"
              >
                CATEGORIES
              </h2>
            </div>
          </div>
        </Sidebar>
        <MainView
          handleLoadMore={[Function]}
          loadMorePending={false}
          pending={true}
        >
          <div
            id="main-view"
          >
            <div
              className="loader"
            />
          </div>
        </MainView>
      </div>
    </Home>
  </Connect(Home)>
</Provider>
`;
